packages/navigation/src/header-height-context/headerHeightContext.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+    PropsWithChildren,
+    createContext,
+    useContext,
+    useEffect,
+    useMemo,
+    useState,
+} from "react";
+import { useHeaderHeight as useReactNavigationHeaderHeight } from "@react-navigation/elements";
+
+const HeaderHeightContext = createContext<{
+    height: number;
+    setHeight?: (h: (height: number) => number) => void;
+}>({
+    height: 0,
+});
+/**
+ * A header height provider that let's you provide the height of the header to components ABOVE React Navigation in the hiearchy.
+ * Useful if you want something to float above all other content, but not block the header.
+ */
+export const HeaderHeightProvider = ({ children }: PropsWithChildren) => {
+    const [height, setHeight] = useState(0);
+
+    const contextValues = useMemo(() => ({ height, setHeight }), [height]);
+    return (
+        <HeaderHeightContext.Provider value={contextValues}>
+            {children}
+        </HeaderHeightContext.Provider>
+    );
+};
+
+/**
+ * Used within mad navigation to provide the header height
+ */
+const _useMadNavigationHeaderHeight = () => {
```

Comment:
What is the intention of using underscore here?
It might be easier to just get this value straight out from the context instead of creating this wrapper.

i.e, using it in HeightSender as
```ts
export const HeightSender = () => {
  const { setHeight } = useContext(HeaderHeightContext);
  // ...
}
```

Reply:
This is true! I normally create a hook for using the context for simplicity, and to make it possibly to hook into the context without exporting the context itself. However since HeightSender is defined in the same file we don't need the middleman. Will fix this!